billy-spain/src/main/java/com/premiumminds/billy/spain/persistence/dao/jpa/DAOESCreditReceiptImpl.java

```diff
@@ -55,16 +55,16 @@ protected Class<JPAESCreditReceiptEntity> getEntityClass() {
         return JPAESCreditReceiptEntity.class;
     }
 
-    @Override
-    public List<ESCreditReceipt> findByReferencedDocument(UID uidCompany, UID uidInvoice) {
+    @Override public List<ESCreditReceipt> findByReferencedDocument(StringID<Business> uidCompany,
+            StringID<GenericInvoice> uidInvoice) {
         QJPAESCreditReceiptEntity creditReceipt = QJPAESCreditReceiptEntity.jPAESCreditReceiptEntity;
         QJPAESCreditReceiptEntryEntity entry = QJPAESCreditReceiptEntryEntity.jPAESCreditReceiptEntryEntity;
         QJPAESGenericInvoiceEntity receipt = QJPAESGenericInvoiceEntity.jPAESGenericInvoiceEntity;
 
         final JPQLQuery<String> invQ = JPAExpressions
             .select(receipt.uid)
             .from(receipt)
-            .where(receipt.uid.eq(uidInvoice.toString()));
+            .where(receipt.uid.eq(uidInvoice.getIdentifier()));
```

Comment:
Shouldn't the QEntity already contain the uid as StringID given the changes in JPABaseEntity in the uid field?

Reply:
That change was left out to reduce the already extended scope of changes and reduce the risk related, changing the type only for the exposed methods. It can be added on future changes if deemed useful or beneficial.